### PR TITLE
add cv=-1 option

### PR DESCRIPTION
**Pull request by gmartinonQM**
# Description

- Simplify unit tests by adding the `noise` option in `make_regression`
- Add the possibility to input `cv=-1` equivalently of `cv=LeaveOneOut()`

## Type of change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

With existing unit tests.

# Checklist:

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `flake8 . --exclude=doc`
- [x] Typing passes successfully : `mypy mapie examples --strict --config-file mypy.ini`
- [x] Unit tests pass successfully : `pytest -vs --doctest-modules mapie`
- [x] Coverage is 100% : `pytest -vs --doctest-modules --cov-branch --cov=mapie --pyargs mapie`
- [x] Documentation builds successfully : `cd doc; make clean; make html`

### CODE REVIEW

File: mapie/estimators.py [hunk 1]
```diff
@@ -190,12 +194,15 @@ def _check_cv(self, cv: Optional[Union[int, BaseCrossValidator]] = None) -> Base
...
-        if isinstance(self.cv, int) and self.cv >= 2:
-            return KFold(n_splits=self.cv)
+        if isinstance(self.cv, int):
+            if self.cv == -1:
+                return LeaveOneOut()
+            if self.cv >= 2:
+                return KFold(n_splits=self.cv)
         if isinstance(self.cv, KFold) or isinstance(self.cv, LeaveOneOut):
             return cv
         raise ValueError("Invalid cv argument. Allowed values are None, int >= 2, KFold or LeaveOneOut.")
```
Comment (vtaquet) [inline mapie/estimators.py:208]: or int == -1

File: mapie/tests/test_estimators.py [hunk 2]
```diff
@@ -23,32 +16,31 @@
...
-
-SEED = 59
-np.random.seed(SEED)
-y_reg = y_reg + np.random.normal(0, 1, 500)
-
+X_reg, y_reg = make_regression(n_samples=500, n_features=10, noise=1.0, random_state=1)
 
 METHODS = ["naive", "base", "plus", "minmax"]
+
+Params = TypedDict("Params", {"method": str, "cv": Optional[Union[int, KFold]]})
```
Comment (vtaquet) [inline mapie/tests/test_estimators.py:23]: Quel est l'avantage d'utiliser un TypedDict par rapport à un Dict classique ?

File: mapie/tests/test_estimators.py [hunk 3]
```diff
@@ -170,15 +163,15 @@ def test_valid_ensemble(ensemble: bool) -> None:
     mapie.fit(X_toy, y_toy)
 
 
-@pytest.mark.parametrize("cv", [-3.14, -2, -1, 0, 1, "cv", DummyRegressor()])
+@pytest.mark.parametrize("cv", [-3.14, -2, 0, 1, "cv", DummyRegressor()])
```
Comment (vtaquet) [inline mapie/tests/test_estimators.py:166]: Que se passe t-il lorsque `cv > X_train.shape[1]` ? Equivalent à un LeaveOneOut ?

**Review by vtaquet — COMMENTED** [review]
Some minor questions are raised.

File: mapie/tests/test_estimators.py (hunk 2 above)
Comment (gmartinonQM) [reply mapie/tests/test_estimators.py:23]: TypedDict ne fait pas crasher mypy
https://mypy.readthedocs.io/en/stable/more_types.html#typeddict

File: mapie/tests/test_estimators.py (hunk 3 above)
Comment (gmartinonQM) [reply mapie/tests/test_estimators.py:166]: Nope, ça lève une erreur sklearn. Je peux le rajouter.

Comment (vtaquet) [reply mapie/tests/test_estimators.py]: Est-ce que ça vaudrait le coup de vérifier le message d'erreur de sklearn dans un test unitaire indépendant ?

**Pull request merged**